engine/database.go

```diff
@@ -0,0 +1,126 @@
+package engine
+
+import (
+	"errors"
+	"fmt"
+	"sync/atomic"
+	"time"
+
+	"github.com/thrasher-/gocryptotrader/database"
+
+	audit "github.com/thrasher-/gocryptotrader/database/repository/audit"
+
+	db "github.com/thrasher-/gocryptotrader/database/drivers/postgres"
+	dbsqlite3 "github.com/thrasher-/gocryptotrader/database/drivers/sqlite"
+
+	auditPSQL "github.com/thrasher-/gocryptotrader/database/repository/audit/postgres"
+	auditSQLite "github.com/thrasher-/gocryptotrader/database/repository/audit/sqlite"
+
+	log "github.com/thrasher-/gocryptotrader/logger"
+)
+
+var (
+	dbConn *database.Database
+)
+
+type databaseManager struct {
+	running  atomic.Value
+	shutdown chan struct{}
+}
+
+func (a *databaseManager) Started() bool {
+	return a.running.Load() == true
+}
+
+func (a *databaseManager) Start() (err error) {
+	if a.Started() {
+		return errors.New("database manager already started")
+	}
+
+	log.Debugln(log.DatabaseMgr, "database manager starting...")
+
+	a.shutdown = make(chan struct{})
+
+	if *Bot.Config.Database.Enabled {
+		if Bot.Config.Database.Driver == "postgres" {
+			dbConn, err = db.Connect()
+			if err != nil {
+				return fmt.Errorf("database failed to connect: %v Some features that utilise a database will be unavailable", err)
+			}
+
+			dbConn.SQL.SetMaxOpenConns(2)
+			dbConn.SQL.SetMaxIdleConns(1)
+			dbConn.SQL.SetConnMaxLifetime(time.Hour)
+
+			err = db.Setup()
+			if err != nil {
+				return err
+			}
+
+			audit.Audit = auditPSQL.Audit()
+		} else if Bot.Config.Database.Driver == "sqlite" {
+			dbConn, err = dbsqlite3.Connect()
+
+			if err != nil {
```

Comment:
Please remove spacing between err and err check

Reply:
done